### ISSUE TITLE
Import bind standard zone file

### ISSUE THREAD

**Issue by madpsy** [ngoduykhanh/PowerDNS-Admin#162]
It might be useful to have the ability to bulk import a zone via an uploaded or pasted in bind compatable zone file. 

There's a few things you could do here, such as support for multiple zone imports, but some way to import even a single zone would be nice. 

Of course there is a possibility of partial imports, e.g. because of RECORDS_ALLOW_EDIT, and generally you wouldn't import NS or SOA record types (although I'm aware pdns Admin doesn't currently have the ability to auto set fixed NS's on zone creation, which would be handy too).

**Comment by ivanfilippov** [comment]
The `pdns-server` package in most distributions comes with a tool called `zone2sql` which will do this for you. 😄

**Comment by trenb** [comment]
Use zone2sql. It works really well and is how we handled our own migration from bind to powerdns.

**Comment by Znuff** [comment]
While `zone2sql` is present on all the systems, that means I have to log on the actual server via SSH and run the SQL commands manually. 

This kind of defeats the purpose of the PowerDNS-Admin. If you're at that, you might as well do everything by SQL/API ...

The ability to import zones directly via PowerDNS-Admin would also help a lot when creating multiple records in a batch. Right now doing this via the PowerDNS-Admin interface is slow and cumberstone.

**Comment by fabiorauber** [comment]
I agree with @Znuff, batch import is a killer feature, and useful in many situations for operators that don't (shouldn't) have access to the PowerDNS server.

**Comment by nikrozman21** [comment]
Any update on this?

**Comment by epipheus** [comment]
oddly zone2sql is nice but it doesn't add anything to the admin for me.

**Comment by swk** [comment]
oddly enough  zone2sql doesnt read actual zone files, it reads named.conf... leaving it where you cant really rapidly import single zones.

**Comment by Znuff** [comment]
Sure you can.

It's literally as simple as `-h` says:

```
# zone2sql --help
syntax:

...

  --zone=...
        Zonefile to parse
  --zone-name=...
        Specify an $ORIGIN in case it is not present
```

So you just need to do:

```
zone2sql --gpgsql --zone=zone-file-name.zone --zone-name='the-domain-name.tld'
```

The `--zone-name` is only required if you don't have a `$ORIGIN` in the zone file.

**Comment by jbe-dw** [comment]
Hello,

This request is stale for too long. Regarding our policy to cleanup this project, we are closing it. This feature request is tracked in https://github.com/PowerDNS-Admin/PowerDNS-Admin/issues/1178

Regards
PowerDNS Admin Team
See https://github.com/PowerDNS-Admin/PowerDNS-Admin/issues/1153 for the policy
